utils/src/test/kotlin/storage/LocalFileStorageTest.kt

```diff
@@ -0,0 +1,152 @@
+ */
+
+package com.here.ort.utils.storage
+
+import com.here.ort.utils.safeMkdirs
+
+import io.kotlintest.shouldBe
+import io.kotlintest.shouldNotThrowAny
+import io.kotlintest.shouldThrow
+import io.kotlintest.specs.WordSpec
+
+import java.io.BufferedReader
+import java.io.File
+
+class LocalFileStorageTest : WordSpec() {
+
+    private fun storage(block: (LocalFileStorage, File) -> Unit) {
+        val directory = createTempDir(javaClass.simpleName)
+        val storage = LocalFileStorage(directory)
+        block(storage, directory)
+    }
+
+    init {
+        "Creating the storage" should {
+            "succeed if the directory exists" {
+                shouldNotThrowAny {
+                    LocalFileStorage(createTempDir(javaClass.simpleName))
+                }
+            }
+
+            "succeed if the directory does not exist and shall be created" {
+                val directory = createTempDir(javaClass.simpleName)
+                val storageDirectory = directory.resolve("create/storage")
+
+                LocalFileStorage(storageDirectory, createDirectory = true)
+
+                storageDirectory.isDirectory shouldBe true
+            }
+
+            "fail if the directory is a file" {
+                val storageDirectory = createTempFile()
+
+                shouldThrow<IllegalArgumentException> {
+                    LocalFileStorage(storageDirectory)
+                }
+            }
+
+            "fail if the directory does not exist and shall not be created" {
+                val directory = createTempDir(javaClass.simpleName)
+                val storageDirectory = directory.resolve("do-not-create/storage")
+
+                shouldThrow<IllegalArgumentException> {
+                    LocalFileStorage(storageDirectory, createDirectory = false)
+                }
+
+                storageDirectory.isDirectory shouldBe false
+            }
+        }
+
+        "Reading a file" should {
+            "succeed if the file exists" {
+                storage { storage, directory ->
+                    val file = directory.resolve("file-exists")
+                    file.writeText("file-exists")
+
+                    val content = storage.read("file-exists").bufferedReader().use(BufferedReader::readText)
+
+                    content shouldBe "file-exists"
+                }
+            }
+
+            "fail if the file does not exist" {
+                storage { storage, _ ->
+                    shouldThrow<IllegalArgumentException> {
+                        storage.read("file-does-not-exist")
+                    }
+                }
+            }
+
+            "fail if the requested path is not inside the storage directory" {
+                storage { storage, _ ->
+                    shouldThrow<IllegalArgumentException> {
+                        storage.read("../file")
+                    }
+                }
+            }
+        }
+
+        "Writing a file" should {
```

Comment:
consider putting a blank line before the assertions

Reply:
Done.